### PR TITLE
private/pedro/status bar search btns disable status

### PR DESCRIPTION
**Pull request by pedropintosilva**
- **Status bar: Enable/disable search buttons**
- **Status bar: Make cancel search icon more subtle**

### CODE REVIEW

File: browser/src/control/Control.StatusBar.js [hunk 1]
```diff
@@ -106,7 +106,9 @@ class StatusBar extends JSDialog.Toolbar {
 		var searchInput = L.DomUtil.get('search-input');
 		if (e.count === 0) {
 			this.enableItem('searchprev', false);
+			this.enableItem('searchprev-button', false);
```
Comment (eszkadev) [inline browser/src/control/Control.StatusBar.js:109]: I think it is a workaround. We should always use "id" in this case "searchprev" only. Probably we have some bad code in the enable/disable implementation

**Comment by pedropintosilva** [comment]
This seems quite serious to don't merge. I prefer to have functional search buttons ASAP then wait for the ideal approach here.

**Review by gokaysatir — APPROVED** [review]
Yep, sorry @pedropintosilva. Let me approve the request. Then we'll have a follow up for even better improvements :)

**Review by eszkadev — CHANGES_REQUESTED** [review]
sorry, I really think it's some fundamental problem we should fix...

**Comment by eszkadev** [comment]
I think we fail here: https://github.com/CollaboraOnline/online/blob/master/browser/src/control/Control.JSDialogBuilder.js#L2942
the example item is: `{type: 'customtoolitem',  id: 'searchprev', command: 'searchprev', text: _UNO('.uno:UpSearch'), enabled: false, pressAndHold: true},`
which has implementation in: https://github.com/CollaboraOnline/online/blob/master/browser/src/control/Control.JSDialogBuilder.js#L2499
which uses `_unoToolButton`
and this leads to https://github.com/CollaboraOnline/online/blob/master/browser/src/control/Control.JSDialogBuilder.js#L2291  where `-button` is present

And that is not the root node. The rule is: we only execute action on the root node of widget (in this example: `div`).
So it is not applied to the button itself. To solve that problem we have helper which synchronizes state with the root node:
eg. `JSDialog.SynchronizeDisabledState(div, [spinfield]);`

**Review by eszkadev — APPROVED** [review]

**Comment by pedropintosilva** [comment]
Cypress (desktop)
```
 Table operations
       Set minimal row height.:
     AssertionError: Timed out retrying after 10000ms: expected '[ <td>, 5 more... ]' not to have attribute 'height'
```

unrelated, re-try

**Comment by pedropintosilva** [comment]
Thanks, but now without my changes and with your commit @eszkadev I now see the next and prev buttons again disabled:

image

I wonder,  and since @gokaysatir was already reviewing, if you @eszkadev would prefer @gokaysatir to take this from you

**Comment by eszkadev** [comment]
yes, @gokaysatir  please check

**Comment by gokaysatir** [comment]
@eszkadev can you have a look at the last commit when you have time? I think "disabled" class usage is being removed from some of the widgets. unotoolbuttons seem to be one of them.

**Review by gokaysatir — APPROVED** [review]

**Comment by eszkadev** [comment]
yes, it seems we use attribute now: https://github.com/CollaboraOnline/online/blob/master/browser/src/control/Control.JSDialogBuilder.js#L2978

**Comment by eszkadev** [comment]
thanks, seems to work now :)

**Pull request merged**